internal/operands/tekton-pipelines/tekton-pipelines.go

```diff
@@ -45,46 +42,48 @@ type tektonPipelines struct {
 
 var _ operands.Operand = &tektonPipelines{}
 
-func New(bundle *tektonbundle.Bundle) *tektonPipelines {
-	tp := &tektonPipelines{
+func New(bundle *tektonbundle.Bundle) operands.Operand {
+	return &tektonPipelines{
 		pipelines:       bundle.Pipelines,
 		configMaps:      bundle.ConfigMaps,
 		roleBindings:    bundle.RoleBindings,
 		serviceAccounts: bundle.ServiceAccounts,
 		clusterRoles:    bundle.ClusterRoles,
 	}
-	return tp
 }
 
 func (t *tektonPipelines) Name() string {
 	return operandName
 }
 
-func (t *tektonPipelines) WatchClusterTypes() []client.Object {
-	return []client.Object{
-		&pipeline.Pipeline{},
-		&v1.ConfigMap{},
-		&rbac.RoleBinding{},
-		&v1.ServiceAccount{},
-		&rbac.ClusterRole{},
+func (t *tektonPipelines) WatchClusterTypes() []operands.WatchType {
+	return []operands.WatchType{
+		{Object: &pipeline.Pipeline{}, Crd: tektonCrd, WatchFullObject: true},
+		{Object: &v1.ConfigMap{}},
+		{Object: &rbac.RoleBinding{}},
+		{Object: &v1.ServiceAccount{}},
+		{Object: &rbac.ClusterRole{}},
 	}
 }
 
-func (t *tektonPipelines) WatchTypes() []client.Object {
+func (t *tektonPipelines) WatchTypes() []operands.WatchType {
 	return nil
 }
 
-func (t *tektonPipelines) RequiredCrds() []string {
-	return requiredCRDs
-}
-
 func (t *tektonPipelines) Reconcile(request *common.Request) ([]common.ReconcileResult, error) {
-	var results []common.ReconcileResult
+	if !request.Instance.Spec.FeatureGates.DeployTektonTaskResources {
```

Comment:
`FeatureGates` pointer can be nil.

Reply:
Done. I've added `request.Instance.Spec.FeatureGates != nil`.